tests/tests_lite/test_connector.py

```diff
@@ -43,25 +44,40 @@
     DDPStrategy,
     DeepSpeedStrategy,
     SingleDeviceStrategy,
+    SingleTPUStrategy,
+    XLAStrategy,
 )
 from lightning_lite.strategies.ddp_spawn import _DDP_FORK_ALIASES
 from lightning_lite.utilities.exceptions import MisconfigurationException
 
 
-def test_accelerator_choice_cpu(tmpdir):
+def test_accelerator_choice_cpu():
     connector = _Connector()
     assert isinstance(connector.accelerator, CPUAccelerator)
     assert isinstance(connector.strategy, SingleDeviceStrategy)
 
 
+@RunIf(tpu=True, standalone=True)
+@pytest.mark.parametrize(["accelerator", "devices"], [("tpu", None), ("tpu", 1), ("tpu", 8), ("auto", 1), ("auto", 8)])
+@mock.patch.dict(os.environ, os.environ.copy(), clear=True)
+def test_accelerator_choice_tpu(accelerator, devices):
+    connector = _Connector(accelerator=accelerator, devices=devices)
+    assert isinstance(connector.accelerator, TPUAccelerator)
+    if devices is None or devices > 1:
+        # TODO: accelerator=tpu, devices=None (default) should imply SingleTPUStrategy
+        assert isinstance(connector.strategy, XLAStrategy)
```

Comment:
This is weird. Probably we need to follow up with a fix to make this consistent. Right now in Lite (and in PL!):

devices=None -> XLAStrategy (spawning one process)
devices=1 -> SingleTPUStrategy (no processes spawn)

Reply:
Should you open an issue or will you resolve it shortly after?

Reply:
Turns out this is expected behavior, but it may change in the future: https://github.com/Lightning-AI/lightning/issues/10606#issuecomment-1017723868